services/subscriber.py

```diff
@@ -2,46 +2,47 @@
 import pytz
 
 from nameko.rpc import rpc
+from elasticsearch_dsl import Search, Q
 from app import es
 
 
 class SubscriberService:
     name = "subscriber_service"
 
     @rpc
-    def get_subscribers(self, countries, hours_whitelist, limit):
+    def get_subscribers(self, targetings, hours_whitelist, limit):
         print("SubscriberService.get_subscribers: getting subscribers")
+        timezones = [tz for tz in pytz.all_timezones
+                     if (datetime
+                         .now(pytz.timezone(tz)).hour
+                         in hours_whitelist)]
+
+        if timezones:
+            targetings.append({
+                "field": "timezone",
+                "operator": "AND",
+                "value": timezones
+            })
         try:
-            timezones = [tz for tz in pytz.all_timezones
-                         if (datetime
-                             .now(pytz.timezone(tz)).hour
-                             in hours_whitelist)]
-            res = es.msearch(body=[
-                {"index": "users"},
-                {
-                    "size": limit,
-                    "query": {
-                        "function_score": {
-                            "query": {
-                                "bool": {
-                                    "must": [
-                                        {"terms": {"country": countries}},
-                                        {"terms": {"timezone": timezones}}
-                                    ]
-                                }
-                            },
-                            "random_score": {},
-                            "boost_mode": "replace"
-                        }
-                    }
-                }
-            ])
-            subscribers = []
-            for row in res['responses'][0]['hits']['hits']:
-                subscriber = row['_source']
-                subscriber['_id'] = row['_id']
-                subscribers.append(subscriber)
-            return subscribers
+            s = Search(using=es, index="users")
+            s.extra(size=limit)
+            logical_operator_mappings = {
```

Comment:
i'm not sure this mapping will be correct,
all targeting must query as `and` clause, it is `must`

what if we use only `must` and `must_not`
and `=` kind of targetings with single value we convert to array and use `terms`